pkg/k8s/init.go

```diff
@@ -228,6 +232,9 @@ func GetNodeSpec() error {
 		}
 
 		node.SetLabels(n.Labels)
+
+		// K8s Node IP is used by BPF NodePort devices auto-detection
+		node.SetK8sNodeIP(n.GetK8sNodeIP())
```

Comment:
What's the implication if we don't hit this case and instead fall through to the else below? 

I feel like the managed clouds were more likely to hit the `}else{` case below, but I'm not particularly familiar with this code.

Reply:
Good concern. The `else` case will be hit, if it cannot retrieve (Cilium)Node object during the first request when both `--k8s-require-ipv{4,6}-pod-cidr` are disabled. I think it is possible only in very busy clusters in which kube-apiserver is overloaded.

If it happens, then only a device with a default route will be able to be selected for BPF NodePort (behavior of the auto-detect in v1.6 and v1.7).

Anyway, I'm going to change `retrieveNodeInformation()` in a way that it keeps waiting for the node object if the auto-detection is required. Mainly, to avoid situations when bpf_host.o is attached to k8s Node device only on some nodes in a busy cluster.

Reply:
Resolved in 8bf878063d6224f9382d5859e3e5bdb7872fb4a0.